File: GGAndroid_2.0/GGDemo/src/main/java/io/ganguo/demo/view/activity/DiffUtilDemoActivity.java
package io.ganguo.demo.view.activity;

import android.content.Context;
import android.content.Intent;

import io.ganguo.demo.R;
import io.ganguo.demo.view.activity.base.GGHFSRVModelActivity;
import io.ganguo.demo.viewmodel.activity.DiffUtilDemoViewModel;
import io.ganguo.viewmodel.common.item.ItemSampleVModel;
import io.ganguo.vmodel.BaseViewModel;
import io.ganguo.vmodel.ViewModelActivity;
import io.ganguo.viewmodel.databinding.IncludeHfSwipeRecyclerBinding;
import io.ganguo.vmodel.ViewModelHelper;

/**
 * DiffUtil 刷新数据工具类 Demo
 * Created by leo on 2018/8/27.
 */
public class DiffUtilDemoActivity extends GGHFSRVModelActivity<IncludeHfSwipeRecyclerBinding, DiffUtilDemoViewModel> {

    public static Intent intentFor(Context context) {
        return new Intent(context, DiffUtilDemoActivity.class);
    }

    @Override
    public DiffUtilDemoViewModel createViewModel() {
        return new DiffUtilDemoViewModel();
    }

    @Override
    public void onViewAttached(DiffUtilDemoViewModel viewModel) {
    }
}

File: GGAndroid_3.0/main/app-mvvm-v1-sample/src/main/java/io/ganguo/demo/view/activity/DiffUtilDemoActivity.kt
package io.ganguo.demo.view.activity

import android.content.Context
import android.content.Intent
import io.ganguo.demo.view.activity.base.GGVModelActivity
import io.ganguo.demo.viewmodel.activity.DiffUtilDemoViewModel
import io.ganguo.viewmodel.core.BaseViewModel
import io.ganguo.viewmodel.databinding.IncludeHfSwipeRecyclerBinding

/**
 * DiffUtil 刷新数据工具类 Demo
 * Created by leo on 2018/8/27.
 */
class DiffUtilDemoActivity : GGVModelActivity<IncludeHfSwipeRecyclerBinding, DiffUtilDemoViewModel>() {

    override fun createViewModel(): DiffUtilDemoViewModel {
        return DiffUtilDemoViewModel()
    }


    override fun onViewAttached(viewModel: DiffUtilDemoViewModel) {
    }

    companion object {

        fun intentFor(context: Context): Intent {
            return Intent(context, DiffUtilDemoActivity::class.java)
        }
    }

}
